client/components/mma/switch/SwitchContainer.tsx

```diff
@@ -114,17 +131,53 @@ const RenderedPage = (props: {
 	user?: MembersDataApiUser;
 	isFromApp?: boolean;
 }) => {
+	const mainPlan = getMainPlan(
+		props.productDetail.subscription,
+	) as PaidSubscriptionPlan;
+	const monthlyOrAnnual = calculateMonthlyOrAnnualFromBillingPeriod(
+		mainPlan.billingPeriod,
+	);
+
 	return (
 		<SwitchPageContainer>
 			<SwitchContext.Provider
 				value={{
 					productDetail: props.productDetail,
 					isFromApp: props.isFromApp,
 					user: props.user,
+					mainPlan,
+					monthlyOrAnnual,
+					supporterPlusTitle: `${monthlyOrAnnual} + extras`,
+					thresholds: getThresholds(mainPlan, monthlyOrAnnual),
 				}}
 			>
 				<Outlet />
 			</SwitchContext.Provider>
 		</SwitchPageContainer>
 	);
 };
+
+function getThresholds(
+	mainPlan: PaidSubscriptionPlan,
+	monthlyOrAnnual: string,
```

Comment:
This might be better as a boolean as that's all it's being used for. (eg. `monthly: boolean`) The actual value of the string isn't being used for anything else in here.